src/huggingface_hub/hub_mixin.py

```diff
@@ -223,24 +227,47 @@ def push_to_hub(
                 private: Whether the model repo should be private (requires a paid huggingface.co account)
             commit_message (:obj:`str`, `optional`, defaults to :obj:`add model`):
                 Message to commit while pushing
+            api_endpoint (:obj:`str`, `optional`):
+                The API endpoint to use when pushing the model to the hub.
+            use_auth_token (``str`` or ``bool``, `optional`, defaults ``None``):
+                huggingface_token can be extract from ``HfApi().login(username, password)`` and is used to authenticate
+                 against the hub (useful from Google Colab for instance).
+            git_user (``str``, `optional`, defaults ``None``):
+                will override the ``git config user.name`` for committing and pushing files to the hub.
+            git_email (``str``, `optional`, defaults ``None``):
+                will override the ``git config user.email`` for committing and pushing files to the hub.
 
         Returns:
             url to commit on remote repo.
         """
         if model_id is None:
             model_id = save_directory
 
-        token = HfFolder.get_token()
+        # The auth token is necessary to create a repo
+        if isinstance(use_auth_token, str):
+            huggingface_token = use_auth_token
+        elif use_auth_token is None and repo_url is not None:
+            # If the repo url exists, then no need for a token
+            huggingface_token = None
+        else:
+            huggingface_token = HfFolder.get_token()
```

Comment:
```suggestion
        if isinstance(use_auth_token, str):
            huggingface_token = use_auth_token
        elif use_auth_token:
            huggingface_token = HfFolder.get_token()
        else:
            huggingface_token = None
```

let's use the same logic as elsewhere no?

Reply:
The issue with using the same logic as elsewhere is that the API becomes weird to use, in order to cover a rare use-case. Namely, the following will never work:

```py
model.push_to_hub("xxx")
```
it needs to be the following:
```py
model.push_to_hub("xxx", use_auth_token=True)
```
I fail to see when a user would want to push to the hub without having an `auth_token`, as it's necessary to create a repo. If the repo already exists and one wants to push to it, then the user already has to specify the `repo_url` to push to.

I think the API is cleaner by having `model.push_to_hub("xxx")` work if you already have an authentication token in your HF folder.

Reply:
Ok